ynr/apps/candidates/tests/test_csv_export.py

```diff
@@ -121,29 +139,119 @@ def test_csv_output(self):
+                ]
+            ),
+        )
+
+    def test_create_csv_management_command_single_election(self):
+        # An empty media directory
+        self.assertEqual(
+            self.storage.listdir(self.storage.base_location), (["images"], [])
+        )
+
+        # We expect a single CSV file
+        call_command("candidates_create_csv", "--election", "2015")
+        self.assertSetEqual(
+            set(self.storage.listdir(".")[1]),
+            set(
+                [
+                    "candidates-2010.csv",
+                    "candidates-2015.csv",
+                    "candidates-local.maidstone.2016-05-05.csv",
+                ]
+            ),
+        )
+
+    def test_create_csv_management_command_single_election_doesnt_exist(self):
+        # An empty media directory, apart from the images dir
+        self.assertEqual(
+            self.storage.listdir(self.storage.base_location), (["images"], [])
+        )
+
+        with self.assertRaises(CommandError):
+            call_command("candidates_create_csv", "--election", "foo")
+
+    def test_create_csv_no_candidates_for_election(self):
+        # An empty media directory, apart from the images dir
+        self.assertEqual(
+            self.storage.listdir(self.storage.base_location), (["images"], [])
+        )
+
+        factories.ElectionFactory.create(
+            slug="2018",
+            name="2018 General Election",
+            for_post_role="Member of Parliament",
+            organization=self.commons,
+        )
+
+        call_command("candidates_create_csv")
+        self.assertSetEqual(
+            set(self.storage.listdir(".")[1]),
+            set(
+                [
+                    "candidates-2010.csv",
+                    "candidates-2015.csv",
+                    "candidates-2018.csv",
+                    "candidates-all.csv",
+                    "candidates-elected-all.csv",
+                    "candidates-local.maidstone.2016-05-05.csv",
+                ]
+            ),
+        )
+
+        empty_file = self.storage.open("candidates-2018.csv").read()
+        self.assertEqual(len(empty_file.splitlines()), 1)
+        self.assertEqual(
+            empty_file,
+            (
+                b"id,name,honorific_prefix,honorific_suffix,gender,birth_date,election,party_i"
+                b"d,party_name,post_id,post_label,mapit_url,elected,email,twitter_username,fac"
+                b"ebook_page_url,party_ppc_page_url,facebook_personal_url,homepage_url,wikiped"
+                b"ia_url,linkedin_url,image_url,proxy_image_url_template,image_copyright,image"
+                b"_uploading_user,image_uploading_user_notes,twitter_user_id,election_date,ele"
+                b"ction_current,party_lists_in_use,party_list_position,old_person_ids,gss_code"
+                b",parlparse_id,theyworkforyou_url,party_ec_id,favourite_biscuits,cancelled_po"
+                b"ll\r\n"
```

Comment:
Representing the expected header row like this feels like its going to be a giant PITA if we ever need to edit it (e.g: because we've added/removed a field).

Could we generate the expected header row based on concatenating https://github.com/DemocracyClub/yournextrepresentative/blob/person_model_changes/ynr/apps/candidates/models/field_mappings.py ?

Reply:
Riiiight, can do, but quite soon the values will be made in some other way (TBC). I'll switch to the field mappings, but we might have to use `PersonIdentifiers` to make some of the list too.